pycop/simulation.py

```diff
@@ -57,13 +62,19 @@ def simu_tstudent(n, m, corrMatrix, nu):
         the simulated sample 
 
     """
+    if not all(isinstance(v, int) for v in [n, m]):
+        raise TypeError("The 'n' and 'm' arguments must both be integer types.")
+    if not isinstance(corr_matrix, np.array):
+        raise TypeError("The 'corr_matrix' argument must be a numpy array.")
+    if not isinstance(nu, float):
```

Comment:
I would consider the possibility that the variable is passed as an integer in the following:
```suggestion
    if not isinstance(nu, (int, float)):
```
For the typing, it seems that we can use _float_ only, as _int_ is implied in that type.